src/FactoryPattern/IProductCar.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace FactoryPattern
+{
+    public interface IProductCar
+    { 
+        string GetCarModel { get; }
```

Comment:
Simplify names. `GetCarModel` to `Model` and `CarColor` to `Color`. When we are using `IProductCar` interface we suppose that the Name field represents Car name so there is no need to write it in prop name.